### PR TITLE
fix: remove code smells from app and bug fixes

### LINKED ISSUE #154: Remove code smells from app
- [ ] Remove any use of `Any`. This is not a dynamic language project, there is no need to return `Any` type from anywhere
- [ ] Use List instead of ArrayList. NEVER return or accept implementation details/concrete classes, use abstractions
- [ ] Remove casts wherever used. There is not a single instance where code written by us will use a cast. As with any use of cast by now in this project, it is just a mistake on the part of developer to first return Any or ArrayList and then cast to appropriate types

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #154

Changes: 
- Removed Any which ultimately removed the extra casting
- All Arraylist are changed to mutableList
- convertImageToLEDHex now does not check for the black value, any value other than transparent is considered black, this removes the restriction of only grayscaled images

### CODE REVIEW

**Comment by adityastic** [comment]
@yashk2000 @yatri1609 @iamareebjamal Please Review

File: app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt [hunk 1]
```diff
@@ -41,8 +41,8 @@ class DrawableAdapter(private val context: Context, private val list: List<Drawa
         }
     }
 
-    fun getSelectedItem(): Any {
-        return if (selectedPosition == -1) -1 else list[selectedPosition]
+    fun getSelectedItem(): DrawableInfo? {
+        return if (selectedPosition == -1) null else list[selectedPosition]
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: Don't return null, either return default drawable or throw exception

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 2]
```diff
@@ -142,7 +143,7 @@ class MessageActivity : AppCompatActivity() {
     private fun setupRecycler() {
         drawableRecyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
 
-        val listOfDrawables = ArrayList<DrawableInfo>()
+        val listOfDrawables = mutableListOf<DrawableInfo>()
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Use listOf

**Comment by adityastic** [comment]
@iamareebjamal Please have a look sir

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 2 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: done

**Pull request merged**